src/api/transfers/handler.js

```diff
@@ -174,30 +187,34 @@ const getTransferById = async function (request, h) {
  * @returns {integer} - Returns the response code 200 on success, throws error if failure occurs
  */
 const fulfilTransferError = async function (request, h) {
+  const { headers, payload, params, dataUri, span } = request
+
+  const metricPrefix = '' // todo: think how to determine fxErrorCallback (by contentType?)
   const histTimerEnd = Metrics.getHistogram(
-    'transfer_fulfil_error',
+    `${metricPrefix}transfer_fulfil_error`,
     'Produce a transfer fulfil error message to transfer fulfil kafka topic',
     ['success']
   ).startTimer()
 
-  const span = request.span
   try {
     span.setTags(getTransferSpanTags(request, Enum.Events.Event.Type.TRANSFER, Enum.Events.Event.Action.ABORT))
-    Logger.isDebugEnabled && Logger.debug(`fulfilTransferError::payload(${JSON.stringify(request.payload)})`)
-    Logger.isDebugEnabled && Logger.debug(`fulfilTransferError::headers(${JSON.stringify(request.headers)})`)
-    Logger.isDebugEnabled && Logger.debug(`fulfilTransfer::id(${request.params.id})`)
+    logger.debug(`fulfilTransferError::payload(${JSON.stringify(request.payload)})`)
+    logger.debug(`fulfilTransferError::headers(${JSON.stringify(request.headers)})`)
+    logger.debug(`fulfilTransfer::id(${request.params.id})`)
```

Comment:
We need to really make sure how these statements work. Because JSON.stringify is very expensive, we want that to be executed only when debug is enabled.

Reply:
Fixed